app/vmalert/alerting.go

```diff
@@ -137,6 +137,7 @@ func (ar *AlertingRule) Exec(ctx context.Context, q datasource.Querier, series b
 		}
 	}
 
+	qFn := func(query string) ([]datasource.Metric, error) { return q.Query(ctx, query) }
```

Comment:
maybe its better to wrap ctx with timeout?

Reply:
We don't timeout value here and rely on limitations applied on prev steps